kratos/processes/time_averaging_process.cpp

```diff
@@ -0,0 +1,270 @@
+              ". Available model parts are: \n";
+        for (std::string model_part_name : r_model_part_names)
+            msg += "     " + model_part_name + "\n";
+
+        KRATOS_ERROR << msg;
+    }
+
+    const ModelPart& r_model_part = mrModel.GetModelPart(mModelPartName);
+
+    for (const std::string& variable_name : mVariableNamesList)
+    {
+        if (KratosComponents<Variable<double>>::Has(variable_name))
+        {
+            const Variable<double>& r_variable =
+                KratosComponents<Variable<double>>::Get(variable_name);
+            KRATOS_ERROR_IF(!r_model_part.HasNodalSolutionStepVariable(r_variable))
+                << r_model_part.Name() << " doesn't have "
+                << r_variable.Name() << " in NodalSolutionStepDataContainer. Please add it as a SolutionStepVariable.";
+        }
+        else if (KratosComponents<Variable<array_1d<double, 3>>>::Has(variable_name))
+        {
+            const Variable<array_1d<double, 3>>& r_variable =
+                KratosComponents<Variable<array_1d<double, 3>>>::Get(variable_name);
+            KRATOS_ERROR_IF(!r_model_part.HasNodalSolutionStepVariable(r_variable))
+                << r_model_part.Name() << " doesn't have "
+                << r_variable.Name() << " in NodalSolutionStepDataContainer. Please add it as a SolutionStepVariable.";
+        }
+        else
+        {
+            KRATOS_ERROR << "Variable " << variable_name << " not found in the scalar and 3d variables list of Kratos.\n";
+        }
+    }
+
+    return 0;
+
+    KRATOS_CATCH("");
+}
+
+void TimeAveragingProcess::ExecuteInitialize()
+{
+    KRATOS_TRY
+
+    ModelPart& r_model_part = mrModel.GetModelPart(mModelPartName);
+    Communicator& r_communicator = r_model_part.GetCommunicator();
+    ModelPart::NodesContainerType& r_nodes = r_communicator.LocalMesh().Nodes();
+
+    mCurrentTime = 0.0;
+
+    std::stringstream msg;
+
+    msg << "Initialized non-historical";
+
+    for (const std::string& variable_name : mVariableNamesList)
+    {
+        if (KratosComponents<Variable<double>>::Has(variable_name))
+        {
+            const Variable<double>& r_variable =
+                KratosComponents<Variable<double>>::Get(variable_name);
+            VariableUtils().SetNonHistoricalVariableToZero(r_variable, r_nodes);
+        }
+        else if (KratosComponents<Variable<array_1d<double, 3>>>::Has(variable_name))
+        {
+            const Variable<array_1d<double, 3>>& r_variable =
+                KratosComponents<Variable<array_1d<double, 3>>>::Get(variable_name);
+            VariableUtils().SetNonHistoricalVariableToZero(r_variable, r_nodes);
+        }
+        msg << " " << variable_name;
+    }
+    msg << " variable(s) in " << mModelPartName << " to store time averaged quantities.\n";
+
+    KRATOS_INFO_IF(this->Info(), mEchoLevel > 0) << msg.str();
+
+    KRATOS_CATCH("");
+}
+
+void TimeAveragingProcess::Execute()
```

Comment:
```suggestion
void TimeAveragingProcess::ExecuteFinalizeSolutionStep()
```
Is it correct defining this function as `Execute()`? Right now this function is called at `ExecuteFinalizeSolutionStep()` and it's the only function called there. I would direclty define this `Execute()` function as `ExecuteFinalizeSoltionStep()`

Also this would mean that from python someone could directly call `self.process.Execute()`, which would enter this function without initializing the process first.

Reply:
Sorry my mistake, I was following my rans implementations in here as well and forgot to change it. Now it is changed as you suggested